### DISCUSSION TITLE
Auto-restart of server on VM crash?

### DISCUSSION THREAD

**Discussion by lukehutch** [serverpod/serverpod#1294] [General]
I have been seeing a *lot* of Dart VM crashes lately. I had no idea the Dart VM was this immature.

I'm only running the server locally at present, so I can auto-restart the server in an infinite for-loop in a bash-script, if I need to. But is there a mechanism for automatically restarting the server on crash, if launched on AWS / GCP?

**Comment by lukehutch** [comment]
Furthermore, I assume that in production, the Serverpod server is AOT-compiled?

**Reply by vlidholt** [reply]
I haven't seen any crashing VMs, maybe it is an issue with a specific Dart version or setup? By default, we use the kernel snapshot compilation for AWS and AOT compilation for GCP.

**Comment by lukehutch** [comment]
Granted most of the crashes have been on the Flutter side, and some of those have been during compilation or hot restarts. I can't remember if I saw crashes on the server side specifically. But I see one or two VM crashes per month overall, which does not give me confidence that the server won't crash.

Any server running on a VM, even a venerable VM like the JVM, *must* have a watchdog or auto-restart mechanism, in my opinion. You simply can't rely on a technology as complex as a VM being bug-free.

**Reply by vlidholt** [reply]
The Terraform scripts and the manual GCP setup will restart offline servers, so that is covered.

**Reply by lukehutch** [reply]
OK. In that case I will close this. Thanks!
